src-java/kilda-model/src/main/java/org/openkilda/model/FeatureToggles.java

```diff
@@ -15,100 +15,246 @@
+                getUseBfdForIslIntegrityCheck(), getFloodlightRoutePeriodicSync(), getFlowsRerouteViaFlowHs(),
+                getFlowsRerouteUsingDefaultEncapType(), getCollectGrpcStats());
+    }
+
+    /**
+     * Defines persistable data of the FeatureToggles.
+     */
+    public interface FeatureTogglesData {
+        Boolean getFlowsRerouteOnIslDiscoveryEnabled();
+
+        void setFlowsRerouteOnIslDiscoveryEnabled(Boolean flowsRerouteOnIslDiscoveryEnabled);
+
+        Boolean getCreateFlowEnabled();
+
+        void setCreateFlowEnabled(Boolean createFlowEnabled);
+
+        Boolean getUpdateFlowEnabled();
+
+        void setUpdateFlowEnabled(Boolean updateFlowEnabled);
+
+        Boolean getDeleteFlowEnabled();
+
+        void setDeleteFlowEnabled(Boolean deleteFlowEnabled);
+
+        Boolean getPushFlowEnabled();
+
+        void setPushFlowEnabled(Boolean pushFlowEnabled);
+
+        Boolean getUnpushFlowEnabled();
+
+        void setUnpushFlowEnabled(Boolean unpushFlowEnabled);
+
+        Boolean getUseBfdForIslIntegrityCheck();
+
+        void setUseBfdForIslIntegrityCheck(Boolean useBfdForIslIntegrityCheck);
+
+        Boolean getFloodlightRoutePeriodicSync();
+
+        void setFloodlightRoutePeriodicSync(Boolean floodlightRoutePeriodicSync);
+
+        Boolean getFlowsRerouteViaFlowHs();
+
+        void setFlowsRerouteViaFlowHs(Boolean flowsRerouteViaFlowHs);
+
+        Boolean getFlowsRerouteUsingDefaultEncapType();
+
+        void setFlowsRerouteUsingDefaultEncapType(Boolean flowsRerouteUsingDefaultEncapType);
+
+        Boolean getCollectGrpcStats();
+
+        void setCollectGrpcStats(Boolean collectGrpcStats);
+    }
 
     /**
-     * Constructor prevents initialization of entityId field.
+     * POJO implementation of FeatureTogglesData.
      */
-    @Builder(toBuilder = true)
-    FeatureToggles(Boolean flowsRerouteOnIslDiscoveryEnabled, Boolean createFlowEnabled, Boolean updateFlowEnabled,
-                   Boolean deleteFlowEnabled, Boolean pushFlowEnabled, Boolean unpushFlowEnabled,
-                   Boolean useBfdForIslIntegrityCheck, Boolean floodlightRoutePeriodicSync,
-                   Boolean flowsRerouteViaFlowHs, Boolean flowsRerouteUsingDefaultEncapType, Boolean collectGrpcStats) {
-        this.flowsRerouteOnIslDiscoveryEnabled = flowsRerouteOnIslDiscoveryEnabled;
-        this.createFlowEnabled = createFlowEnabled;
-        this.updateFlowEnabled = updateFlowEnabled;
-        this.deleteFlowEnabled = deleteFlowEnabled;
-        this.pushFlowEnabled = pushFlowEnabled;
-        this.unpushFlowEnabled = unpushFlowEnabled;
-        this.useBfdForIslIntegrityCheck = useBfdForIslIntegrityCheck;
-        this.floodlightRoutePeriodicSync = floodlightRoutePeriodicSync;
-        this.flowsRerouteViaFlowHs = flowsRerouteViaFlowHs;
-        this.flowsRerouteUsingDefaultEncapType = flowsRerouteUsingDefaultEncapType;
-        this.collectGrpcStats = collectGrpcStats;
+    @Data
+    @Builder
```

Comment:
Right, it can be just a constructor. But IDEA will not help with pulling constructor arguments from FeatureTogglesDataImpl up to FeatureToggles constructor. Developers have to do it manually. And idea behind usage of builders here - to avoid mixing up the arguments (with a builder, you see the names)